.github/workflows/style_check.yml

```diff
@@ -2,47 +2,32 @@ name: Style check
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
 
 on:
   # Trigger on pull requests.
   pull_request:
 
-  # Trigger on pushes to the mainline branches and version tags. This prevents building commits
+  # Trigger on pushes to the mainline branches. This prevents building commits
   # twice when the pull request source branch is in the same repository.
   push:
     branches:
       - "master"
 
-  # Trigger on request.
-  workflow_dispatch:
-
 jobs:
-  style_checks:
-    name: Run ${{ matrix.name }}
+  pre-commit:
+    name: Run pre-commit
     runs-on: ubuntu-latest
-    container:
-      image: glotzerlab/ci:2021.05-clang10_py38
-      options: -u 0
-    strategy:
-      matrix:
-        include:
-          - name: clang_format
-            command: ./run-clang-format.py -r .
-          - name: flake8
-            command: flake8
-          - name: yapf
-            command: yapf -d -r .
-
     steps:
-      - uses: actions/checkout@v2.3.4
-      - name: Execute ${{ matrix.name }}
-        run: ${{ matrix.command }}
+    - uses: actions/checkout@v2.3.4
+    - uses: actions/setup-python@v2.2.2
+    - uses: pre-commit/action@v2.0.2
+      with:
+        extra_args: --all-files --hook-stage manual
```

Comment:
Does this run all the checks, or _just_ the manual ones?

Reply:
``--all-files --hook-stage manual`` runs all the checks.
You can use ``--all-files --hook-stage manual clang-format`` to run just clang-format.